elyra/pipeline/processor_local.py

```diff
@@ -112,17 +129,75 @@ def _sort_operation_dependencies(operations_by_id: dict, ordered_operations: lis
                                                                         parent_operation)
             ordered_operations.append(operation)
 
-    def _get_envs(self, operation: Operation) -> Dict:
-        """Operation stores environment variables in a list of name=value pairs, while
-           subprocess.run() requires a dictionary - so we must convert.  If no envs are
-           configured on the Operation, the existing env is returned, otherwise envs
-           configured on the Operation are overlayed on the existing env.
-        """
-        envs = os.environ.copy()
-        for nv in operation.env_vars:
-            nv_pair = nv.split("=")
-            if len(nv_pair) == 2:
-                envs[nv_pair[0]] = nv_pair[1]
-            else:
-                self.log.warning(f"Could not process environment variable entry `{nv}`, skipping...")
-        return envs
+
+class OperationProcessor(ABC):
+
+    def __init__(self):
+        self.log = log.get_logger()
+
+    @abstractmethod
+    def operation_name(self) -> str:
+        pass
+
+    @abstractmethod
+    async def process(self, operation: Operation, filepath: str):
```

Comment:
Per the previous comment, I think `operation` should be the only parameter here.  Everything comes from the `operation` that an OperationProcessor needs.

Reply:
see the previous response about root-dir